### ISSUE TITLE
Size Limit when installing on TFS 2018 Server

### ISSUE THREAD

**Issue by ejohn20** [dependency-check/azuredevops#7]
Copied from VSMarketplace: 

Jeffrey Rempel: How do you install this on a TFS 2018 on-premise server? I try to upload the extension however I get an error stating "The files being added total 35.7 MB which exceeds the maximum total size allowed to be uploaded (25 MB)."

**Comment by ejohn20** [comment]
Jeff - great question. I'm not sure how this works for on-premise servers. We had to request an increase in the package size from the marketplace support team in order to publish the extension. They would probably be the best resource to reach out to.

**Comment by ejohn20** [comment]
Jeffrey Rempel: I have opened a support case with the Azure DevOps / TFS team. I'll post back here if I get a resolution from them.

**Comment by Friesens-JeffreyR** [comment]
SOLVED:
The support team at Microsoft came back with a query that I needed to run on my TFS SQL server in the configuration database. After I ran this I was able to upload the extension.

`EXEC dbo.prc_SetRegistryValue @partitionId = 1, @key = '#\Configuration\Service\Gallery\LargeExtensionUpload\MaxPackageSizeMB\', @value = 128, @identityName = null`

**Comment by Eyob-T** [comment]
We are using Azure DevOps Server 2019 and ran the query mentioned by @Friesens-JeffreyR  but we were still receiving the same error - _"The files being added total 35.7 MB which exceeds the maximum total size allowed to be uploaded (25 MB)."_. Are there any other solutions ? - been digging online and found the same problem but did not yet find a solution

**Comment by lanphiergm** [comment]
We are experiencing the same issue on 2019.1 and this solution does not work for the latest version in the marketplace (5.2.1.2). 

Poking around in the VSIX file, it appears that almost all of the JAR files appear in there twice: once under the lib folder and once under the repo folder. Without all the duplication, the file would be well under the maximum size and negate the need for workarounds. I tried manually pruning it down some to get it under the limit but it appears to be using some from one folder and some from the other in a pattern I couldn't identify quickly. 

Could you please fix this?

**Comment by ejohn20** [comment]
It sounds like MS has changed the query / method for enabling large extensions. 

Short term - Does anyone have support w/ MS to find out how to increase the size on the 2019.1 version?

Long term - I think we can probably curl the installer package down from artifactory on the fly as the extension loads instead of bundling into the installer. This would eliminate the problem all together and shouldn't take too much time.

**Comment by Eyob-T** [comment]
UPDATE: We were able to update the registry value with the same query mentioned above but this time we did the actual query by logging into the server. The extension is now installed - version 5.2.1.2.

Side note: Before we ran the query we searched for the `_key_` parameter `(#\Configuration\Service\Gallery\LargeExtensionUpload\MaxPackageSizeMB\)` in the configuration DB but it did not show up in the _RegistryItems_ table until after we actually ran the query.
